### DISCUSSION TITLE
Modularizar o gulp

### DISCUSSION THREAD

**Discussion by paulinhoerry** [frontendbr/forum#312] [Discussão]
Galera, queria saber a opinião de vocês a respeito da modularização das tasks Gulp. 
Atualmente estou num projeto que tem uma quantidade considerável de tasks - precisamente 12 tasks - e cada task tem seu arquivo JS separado. Acontece que as vezes buga e eu tenho que reiniciar o gulp, daí tou pensando em desmodularizar. A pergunta é: **Vocês acham viável modular as tasks do Gulp?**

**Comment by woliveiras** [comment]
O fato de bugar deve ser melhor analisado viu @paulinhoerry.

Pode ser alguma task específica travando o processamento, outra task mais lenta que esteja demorando tanto que você desiste e reinicia o Gulp, etc.

Sobre modularizar, acho super útil. Quando você precisa deletar uma tarefa basta apagar o arquivo referente a mesma. Quando você precisa modificar algo é bem mais rápido de encontrar, pois você sabendo o nome da tarefa, sabe qual deve ser o arquivo com suas especificações.

Um esquema que gosto de usar pra Modularizar é esse aqui: 

https://github.com/woliveiras/modularize-your-gulpfile

E tem esse esquema bem da hora do @fernandofleury: https://medium.com/@fernandofleury/uma-abordagem-para-modularizar-seu-gulpfile-fc513cac5f5f

**Comment by felipefialho** [comment]
Cara,

Já tentei usar.

Mas depois que parei pra pensar analisei que não fazia sentido. 
Acho muito mais prático ter todas as tasks dentro do Gulpfile. 

Não consigo compreender qual seria a vantagem principal de modularizar, já que na questão de facilidade de encontrar e fazer modificações, é só dar um find dentro do Gulpfile.

**Comment by theandersonn** [comment]
Eu prefiro tudo recortadinho.

É uma questão de gosto [sou um pouco viciado em tudo recortado], prefiro uma leitura e compreensão modularizada de cada task.

Quanto ao bug, não é por estar separado. As tarefas devem estar em conflito, tem que debugar e passar um pipe pra sinalizar quando uma termina e a outra começa deve resolver.

Exemplo de como estou organizando atualmente: (Vez por outra eu mudo algo)
https://github.com/theandersonn/gulp-bootstrap

**Comment by fernandofleury** [comment]
Eu faço [assim](https://github.com/fernandofleury/fernandofleury.github.io/blob/feat/blog/gulpfile.js) hoje em dia

**Comment by paulinhoerry** [comment]
@woliveiras Então, devia ser bug mesmo, mas de qualquer forma não tava fazendo muito sentido pra mim. Eu modularizei de uma maneira muito embaraçosa, essa que você me apresentou é bem mais limpa. ahahah

@LFeh Foi por não achar sentindo também que eu desmodularizei tudo. kkkkkkkkkk

@theandersonn então, eu nem me atentei a debugar pra achar o erro. Porquê tipo, quando eu passava um pouco mais de 1hr, o watch do gulp já não funcionava mais, aí eu tinha que reiniciar.

@fernandofleury Muito bom, Fernando. Eu li teu artigo no Medium, vou ver se encaixa em algum projeto futuro. Gratidão pela informação compartilhada, Fernando!

**Comment by felipefialho** [comment]
@fernandofleury A sua ideia é boa. Daí você pode sempre carregar a pasta com todos os módulos mas só chamar o que for necessário pra cada projeto, certo?

**Comment by fernandofleury** [comment]
@paulinhoerry eu não sigo mais o modelo do artigo. Olha o link que eu comentei.

@LFeh posso até versionar as tasks mais genéricas se necessário, mas a ideia é algo assim.

**Comment by nicklima** [comment]
@fernandofleury, eu achei bem interessante a modulação do teu Gulpfile. Parabéns pelo artigo. Tentar montar algo usando seu modelo como base.
